File: SQL Server bulk insert and bulk updates - VB.NET/[VB]-SQL Server bulk insert and bulk updates - VB.NET/VB/DataGridView_csv_Simple/DataOperations.vb
Imports System.Data.SqlClient
''' <summary>
''' Responsible for retrieving data just inserted from PersonExporter.PersonExporter method
''' which in turn is presented in a child form to display and/or modify the data.
''' </summary>
Public Class DataOperations
    Private ConnectionString As String = "Data Source=KARENS-PC;Initial Catalog=BulkCopyDatabaseCodeSample;Integrated Security=True"
    '        
    '         * Note TOP 8, this will wipe out all but 8 records in the merge operation so if you want
    '         * to see all 5,000 records affected remove the TOP 8 clause. If you leave "as is" you can
    '         * remove it then run the code again to re-populate the initial 5,000 plus records.
    '         
    Private selectQuery As String = "SELECT TOP 8 Id,FirstName,LastName,Gender,BirthDay FROM BulkCopyDatabaseCodeSample.dbo.Person"
    Public Function GetPeople() As DataTable
        Dim dt = New DataTable()
        Using cn As SqlConnection = New SqlConnection With {.ConnectionString = ConnectionString}
            Using cmd As New SqlCommand() With {.Connection = cn, .CommandText = selectQuery}
                cn.Open()
                dt.Load(cmd.ExecuteReader())
            End Using
        End Using

        Return dt
    End Function
End Class

File: Show and edit date only and time only in a DataGridView/[C#][VB]-Show and edit date only and time only in a DataGridView/C# and VB/WindowsFormsApplication1/Classes/DataOperations.cs
using System.Data;
using System.Data.SqlClient;

public class DataOperations
{
    /// <summary>
    /// Replace with your SQL Server name
    /// </summary>
    private string Server = "KARENS-PC";
    /// <summary>
    /// Database in which data resides, see 
    /// CreateDatabaseTableAndData.sql
    /// </summary>
    private string Catalog = "DataGridViewDateTimeColumns";
    /// <summary>
    /// Connection string for connecting to the database
    /// </summary>
    private string ConnectionString = "";


    public DataOperations()
    {
        ConnectionString = $"Data Source={Server};Initial Catalog={Catalog};Integrated Security=True";
    }
    public DataTable GetRows()
    {
        DataTable dt = new DataTable();
        using (SqlConnection cn = new SqlConnection { ConnectionString = ConnectionString })
        {
            using (SqlCommand cmd = new SqlCommand { Connection = cn })
            {
                cmd.CommandText = "SELECT Identifier,RoomNumber,CheckIn,CheckOut FROM Table1";

                cn.Open();
                dt.Load(cmd.ExecuteReader());

            }
        }

        return dt;

    }
}